lib/client/conntest/database/mysql.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package database
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/go-mysql-org/go-mysql/client"
+	"github.com/go-mysql-org/go-mysql/mysql"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/lib/srv/db/common"
+)
+
+// MySQLPinger implements the DatabasePinger interface for the MySQL protocol.
+type MySQLPinger struct{}
+
+// Ping connects to the database and issues a basic select statement to validate the connection.
+func (p *MySQLPinger) Ping(ctx context.Context, ping PingParams) error {
+	if err := ping.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	var nd net.Dialer
+	addr := fmt.Sprintf("%s:%d", ping.Host, ping.Port)
+	conn, err := client.ConnectWithDialer(ctx, "tcp", addr,
+		ping.Username,
+		"", // no password, we're dialing into a tunnel.
+		ping.DatabaseName,
+		nd.DialContext,
+	)
+	if err != nil {
+		// convert the error from MySQL client since it can be wrapped in a "Causer".
+		// The MySQL engine in the agent already does this, but we need it here because
+		// the error is from the MySQL client.
+		return trace.Wrap(common.ConvertError(err))
+	}
+
+	defer func() {
+		if err := conn.Close(); err != nil {
+			logrus.WithError(err).Info("failed to close connection in MySQLPinger.Ping")
+		}
+	}()
+
+	if err := conn.Ping(); err != nil {
```

Comment:
Does the Ping validate the Database name as well?
Or just the connection?

Sorry, I'm not familiar with how the mysql protocol works

Reply:
In Teleport MySQL integration mysql db name is optional and a user can connect to the MySQL withough specifying any database name. After connection if can freely switch between databases using MySQL CLI commands. The Postgres integrant is quite different because the database name needs to be known during connection time.

Reply:
Thank you :pray: 

We might need to change this in the UI then, not sure if it allows for an empty DB Name
cc @kimlisa @ryanclark

Reply:
Yeah I could not get the UI to allow me to use a blank db name.

Reply:
should i only allow blank db name for certain databases? or in general allow it?

Reply:
allow in general I think. The connection tester api should handle that.